pan-domain-auth-play_2-4-0/src/main/scala/com/gu/pandomainauth/action/Actions.scala

```diff
@@ -119,17 +119,12 @@ trait AuthActions extends PanDomainAuth {
     GoogleAuth.validatedUserIdentity(token).map { claimedAuth =>
       val authedUserData = existingCookie match {
         case Some(c) => {
-          val existingAuth = try {
-            CookieUtils.parseCookieData(c.value, settings.publicKey)
-          } catch {
-            case NonFatal(e) =>
-              LegacyCookie.parseCookieData(c.value, settings.secret)
-          }
+          val existingAuth = CookieUtils.parseCookieData(c.value, settings.publicKey)
           Logger.debug("user re-authed, merging auth data")
 
           claimedAuth.copy(
             authenticatingSystem = system,
-            authenticatedIn = Set(system),
+            authenticatedIn = existingAuth.authenticatedIn ++ Set(system),
```

Comment:
Here, we merge in the existing systems - which should reduce the number of cookie re-writes between systems.